lib/SILGen/SILGenPattern.cpp

```diff
@@ -2555,19 +2557,77 @@ void SILGenFunction::usingImplicitVariablesForPattern(Pattern *pattern, CaseStmt
   variableSwapper();
 }
 
+static void emitDiagnoseOfUnexpectedEnumCaseValue(SILGenFunction &SGF,
+                                                  SILLocation loc,
+                                                  ManagedValue value,
+                                                  const EnumDecl *enumDecl) {
+  ASTContext &ctx = SGF.getASTContext();
+  auto diagnoseFailure = ctx.getDiagnoseUnexpectedEnumCaseValue(nullptr);
+  if (!diagnoseFailure) {
+    SGF.B.createBuiltinTrap(loc);
+    return;
+  }
+
+  assert(enumDecl->isObjC());
+  assert(enumDecl->hasRawType());
+
+  // Get the enum type as an Any.Type value.
+  SILType metatypeType = SILType::getPrimitiveObjectType(
```

Comment:
You’re using getLoweredType() below, using it here will allow you to pass a non-canonical metatypes without representation. A bit simpler...

Reply:
I happen to already have canonical metatypes since they're coming from a SIL value, but I'll use getLoweredType consistently anyway.